annotations/views.py

```diff
@@ -29,6 +29,17 @@
 from core.utils import XLSX
 
 
+class CheckOwnerOrStaff(UserPassesTestMixin):
+    """Limits access only to creator of annotation or staff users"""
+
+    def test_func(self):
+        return self.get_object().annotated_by == self.request.user or \
+            self.request.user.is_staff or self.request.user.is_superuser
+
+    def handle_no_permission(self):
```

Comment:
I'm not sure why you override this function, when `test_func` is false it should raise a PermissionDenied, right? And then you don't need the `access_denied.html` template, am I correct?

Reply:
correct, but then I ended up getting an ugly 403 forbidden page from the web server

Reply:
Haha, that's true. Maybe we could then implement the custom error view like https://docs.djangoproject.com/en/2.2/topics/http/views/#customizing-error-views? So adding a `handler403` in `views.py`.

Reply:
yes, that's a good idea. I'll do that.